### PR TITLE
Improve the speed of using COMPlus_JitTimeLogCsv

### PR DESCRIPTION
**Pull request by BruceForstall**
I noticed that setting this and running a superpmi replay
was super slow. (Btw, note that you must use superpmi `--sequential`
in the scenario to avoid parallel writes to the log file).

This change fixes this in a number of ways:
1. The `CycleTimer::CyclesPerSecond()` function does non-trivial
computation to compute its answer. So cache the result.
2. It's expensive to constantly open and close the output file
(once per function). Partially, this is because anti-virus seems to
get involved. So open it once, and keep it open until the JIT
process is shut down.
3. Avoid calling `eeGetMethodFullName()` in DEBUG builds. (Note that
JitTimeLogCsv collects interesting information beyond just per-phase
timings that are also useful for extracting from a CSV file.)

### CODE REVIEW

File: src/coreclr/jit/compiler.cpp [hunk 1]
```diff
@@ -8233,8 +8244,14 @@ void JitTimer::PrintCsvMethodStats(Compiler* comp)
...
 
-    // eeGetMethodFullName uses locks, so don't enter crit sec before this call.
-    const char* methName = comp->eeGetMethodFullName(comp->info.compMethodHnd);
+// eeGetMethodFullName uses locks, so don't enter crit sec before this call.
+#if defined(DEBUG) || defined(LATE_DISASM)
+    // If we already have computed the name because for some reason we're generating the CSV
+    // for a DEBUG build (presumably not for the time info), just re-use it.
+    const char* methName = comp->info.compFullName;
+#else
+    const char*          methName  = comp->eeGetMethodFullName(comp->info.compMethodHnd);
```
Comment (BruceForstall) [inline src/coreclr/jit/compiler.cpp:8253]: Sometimes jit-format does crazy things...

**Comment by BruceForstall** [comment]
@dotnet/jit-contrib PTAL

File: src/coreclr/jit/compiler.cpp [hunk 2]
```diff
@@ -8186,40 +8192,45 @@ void JitTimer::PrintCsvHeader()
...
-            fprintf(fp, "\"IL Bytes\",");
-            fprintf(fp, "\"Basic Blocks\",");
-            fprintf(fp, "\"Min Opts\",");
-            fprintf(fp, "\"Loops Cloned\",");
+            fprintf(s_csvFile, "\"Method Name\",");
+            fprintf(s_csvFile, "\"Assembly or SPMI Index\",");
+            fprintf(s_csvFile, "\"IL Bytes\",");
+            fprintf(s_csvFile, "\"Basic Blocks\",");
+            fprintf(s_csvFile, "\"Min Opts\",");
+            fprintf(s_csvFile, "\"Loops\",");
```
Comment (BruceForstall) [inline src/coreclr/jit/compiler.cpp:8212]: This is a slight functional change: adding the number of loops in a function to the CSV

**Review by sandreenko — APPROVED** [review]
LGTM, thank you!

**Pull request merged**